Runtime/Api/V1/GameEvents/GameEventApi.cs

```
@@ -0,0 +1,40 @@
+using System;
+using System.Collections.Generic;
+using UnityEngine.Networking;
+
+namespace PlayerZero.Api.V1
+{
+    public class GameEventApi : WebApi
+    {
+        private const string Resource = "public/events";
+        
+        /// <summary>
+        /// Sends a generic game event to the analytics server asynchronously.
+        /// </summary>
+        /// <typeparam name="T">The type of the payload.</typeparam>
+        /// <param name="request">The event data to send.</param>
+        public void SendGameEvent<T>(T request)
+        {
+            try
+            {
+                var apiRequest = new ApiRequest<T>()
+                {
+                    Url = $"{Settings.ApiBaseUrl}/v1/{Resource}",
+                    Method = UnityWebRequest.kHttpVerbPOST,
+                    Headers = new Dictionary<string, string>()
+                    {
+                        { "Content-Type", "application/json" },
+                    },
+                    Payload = request
+                };
+                Dispatch<GameEventResponse, T>(apiRequest);
+
+            }
+            catch (Exception e)
+            {
+                Console.WriteLine(e);
+                throw;
```

I like that you've tried catched the error and then logged it to the console, but maybe it shouldn't throw afterwards? Seems like if analytics fails then we wouldn't want to break the whole integration?

---

have removed thet try catch

---

Is that right though? I feel like analytics shouldn't cause an application to break...

---

makes sense, analytics shouldn't block the game from working, events will just fail silently